### PR TITLE
Tech 376

### PR DESCRIPTION
**Pull request by sujnesh**

### CODE REVIEW

File: app/controllers/submissions_controller.rb [hunk 1]
```diff
@@ -106,6 +106,10 @@ def show
   end
 
   def new
+    if @challenge.min_team_participants > 1 && (@challenge.min_team_participants > get_team_participants.count)
+      redirect_to @challenge,
```
Comment (skbly7) [inline app/controllers/submissions_controller.rb:110]: `redirect_or_json` need to be used

File: app/controllers/concerns/challenge_masthead.rb [hunk 2]
```diff
@@ -8,6 +8,7 @@ def challenge_masthead_actions(actions)
         before_action :set_challenge_rounds, only: actions
         before_action :set_vote, only: actions
         before_action :set_follow, only: actions
+        before_action :redirect_if_hidden, only: actions
```
Comment (skbly7) [inline app/controllers/concerns/challenge_masthead.rb:11]: Not sure why it is here? Should it be part of Discussions controller?

**Review by skbly7 — CHANGES_REQUESTED** [review]

File: app/policies/challenge_policy.rb [hunk 3]
```diff
@@ -159,6 +160,13 @@ def clef_participant_registered?(challenge)
     end
   end
 
+  def get_team_participants_count
+    team = participant.teams.where(challenge_id: @record.id).first
```
Comment (skbly7) [inline app/policies/challenge_policy.rb:164]: what if there is no team?

Comment (sujnesh) [inline app/policies/challenge_policy.rb:164]: Will break if no team

Comment (skbly7) [reply app/policies/challenge_policy.rb:164]:
```
if team.present?
    participants = team.team_participants.map(&:participant) if team.present?
    participants = [participant] if participants.blank?
    return participants.count
end
return 1
```

**Review by skbly7 — APPROVED** [review]

**Pull request merged**